h2/stream.py

```diff
@@ -787,15 +789,14 @@ def push_stream_in_band(self, related_stream_id, headers, encoder):
         # Because encoding headers makes an irreversible change to the header
         # compression context, we make the state transition *first*.
 
-        # This does not trigger any events.
```

Comment:
Note that push promise is going to put some tricky assumptions to you: in particular, the client/server dichotomy no longer applies quite so cleanly. The headers in a push-promise frame are effectively *client* headers sent *by* a server.